plugins/inputs/docker/docker.go

```diff
@@ -221,14 +221,18 @@ func (d *Docker) gatherContainer(
 		cname = strings.TrimPrefix(container.Names[0], "/")
 	}
 
-	// the image name sometimes has a version part.
-	//   ie, rabbitmq:3-management
-	imageParts := strings.Split(container.Image, ":")
-	imageName := imageParts[0]
+	// the image name sometimes has a version part, or a private repo
+	//   ie, rabbitmq:3-management or docker.someco.net:4443/rabbitmq:3-management
+	imageName := ""
 	imageVersion := "unknown"
-	if len(imageParts) > 1 {
-		imageVersion = imageParts[1]
+	imageVersionMarkerIndex := strings.LastIndex(container.Image, ":")
```

Comment:
I know it goes against other languages but general Go style dictates the use of short variable names. Since this is an index that is scoped just within these 3 lines of code, just call it `i` (instead of `imageVersionMarkerIndex`). You can add a comment explaining what it is exactly if you like.